src/Microsoft.Dnx.Runtime/Internal/ConcurrencyUtilities.cs

```diff
@@ -65,5 +71,43 @@ public static void ExecuteWithFileLocked(string filePath, Action<bool> action)
             // should never get here
             throw new TaskCanceledException($"Failed to acquire semaphore for file: {filePath}");
         }
+
+#if DNXCORE50
+        private class Semaphore
+        {
+            private readonly System.Threading.Semaphore _semaphore;
```

Comment:
Yes, this class itself has the name `Semaphore` but we want to use the one from standard library.